### ISSUE TITLE
Crash on exiting with Python 3.6

### ISSUE THREAD

**Issue by pfmoore** [ipython/ipython#10112]
I've just installed:

1. The release version of Python 3.6.0 (Windows 64-bit, from the executable installer)
2. IPython, via ```pip install ipython``` (plus some other packages at the same time)

Now, if I start IPython via ```ipython``` and immediately use ```Ctrl-D``` to exit, I get the "Do you really want to exit" prompt, and when I choose "y", I get "Python has stopped working".

Event Viewer doesn't show much useful in the way of details, but in case it helps, what I can see is:

```
System 

  - Provider 

   [ Name]  Application Error 
 
  - EventID 1000 

   [ Qualifiers]  0 
 
   Level 2 
 
   Task 100 
 
   Keywords 0x80000000000000 
 
  - TimeCreated 

   [ SystemTime]  2016-12-30T12:06:32.000000000Z 
 
   EventRecordID 138677 
 
   Channel Application 
 
   Computer mypc 
 
   Security 
 

- EventData 

   python.exe 
   3.6.150.1013 
   585cdba5 
   ntdll.dll 
   6.1.7601.23569 
   57f7c06e 
   c0000374 
   00000000000bf3e2 
   205c 
   01d2629525cb1311 
   c:\users\uk03306\appdata\local\programs\python\python36\python.exe 
   C:\Windows\SYSTEM32\ntdll.dll 
   66c4cfea-ce88-11e6-813a-5ce0c5b8d4cb 
```

**Comment by pfmoore** [comment]
This appears to be due to the use of ```win-unicode-console``` which is no longer needed for Python 3.6 (as Python supports Unicode directly in the console now). The crash comes when disabling ```win-unicode-console```. I would suggest simply not using the module in 3.6+.

As a simple test, I modified ```IPython/terminal/interactiveshell.py``` as follows:

```python
    def enable_win_unicode_console(self):
        # Add the following 2 lines
        if sys.version_info >= (3, 6):
            return
```

and this stopped the crash.

**Comment by takluyver** [comment]
I thought I'd already added that check; maybe it's not in a release version
yet. If not, we should do a bug fix release soon.

**Comment by pfmoore** [comment]
Ah, didn't think to check the github version, sorry. I have 5.1.0.

**Comment by pfmoore** [comment]
... and yes, that precise check is in github. Please return the keys to Guido's time machine when you're finished with it :-)

**Comment by xoviat** [comment]
@pfmoore 

> Tristan:
> 
> The Pypiwin32 package does not currently support Python 3.6. I understand that you may not have time to resolve this, but it is very difficult to submit pull requests for a  project that doesn’t have a repository on a reasonable service (I would not consider sourceforge to be reasonable). In addition, interest from Mark Hammond has not been forthcoming. 
> 
> To solve this problem, I have created the pywin32 organization on GitHub and have invited the package maintainers as members. You can even remove me from the members if you want. All I care about is having a reasonable place where people can look at the code and submit pull requests if they need to resolve a particular problem. In this case, I am hoping that you will take control of the organization and then either repair setup.py so that it works on Python 3.6 or allow me to submit a pull request.
> 
> Regards,
> 
> xoviat
>

**Comment by pfmoore** [comment]
@xoviat I'm not sure how your comment relates to the current issue, but the pypiwin32 project is simply rebuilds of the pywin32 project, which is hosted on Sourceforge and does not provide wheels. Pywin32 provides wininst installers for 3.6, which can be converted to wheels using ```wheel convert``` or Christoph Gohlke hosts wheels on his site.

The pypiwin32 project was set up by glyph to offer a source for pywin32 wheels which were usable from pip. His original builds were rebuilds, and I added some extra builds generated via ```wheel convert```.

I don't see much advantage in a Github organisation - requests against pywin32 should be directed to their project infrastructure at sourceforge. I do *not* want to try to offer support for pypiwin32 independently of pywin32 (I can't speak for the other maintainers on this, though). What I might do is try to find some time to ```wheel convert``` the pywin32 220 builds from sourceforge and add them to the pypiwin32 project, though. No promises, though - IIRC glyph had some concerns about simple conversion, so I'll need to check with him as to whether he'd be OK with that.

**Comment by xoviat** [comment]
> I don't see much advantage in a Github organisation - requests against pywin32 should be directed to their project infrastructure at sourceforge. I do not want to try to offer support for pypiwin32 independently of pywin32 (I can't speak for the other maintainers on this, though).

@pfmoore The reason that I am attempting to move in this direction is that there the infrastructure at sourceforge is inadequate. The source build released is not compatible with PIP or setuptools standards and no one can do anything about it because there is no way to submit pull requests and there is no version control system at the sourceforge project. The reason that I contacted you here is that you are on the package maintainer's list (I certainly cannot upload a new .tar.gz to resolve the issues) and pypiwin32 has become the de facto standard because no one wants to head over to sourceforge. 

A perfect example of the problems caused by this is the following response by Mark Hammond:

> This is a problem with the pypiwin32 package, which isn't part of this project. setup.py must use old-style print as it needs to work on python 2.x. pypiwin32 should be calling setup3.py.

I would not ask you to resolve these problems. But what I am saying is that right now **no one** can resolve these problems. No one can even checkout the code to resolve these problems. They're not difficult to fix, they simply cannot fixed be with the current organizational structure.

**Comment by minrk** [comment]
#9946 is the fix, currently in master and marked for 5.2. @takluyver since this is needed for Python 3.6 on Windows, should we make a 5.1.1 with just this? If so, any others for a 5.1.1?

**Comment by takluyver** [comment]
Closing as this is fixed in master. I'm trying to push towards a 5.2 release.
